modules/scheduler/scheduler/src/main/java/org/motechproject/scheduler/builder/SchedulableJobBuilder.java

```diff
@@ -0,0 +1,113 @@
+package org.motechproject.scheduler.builder;
+
+import org.joda.time.DateTime;
+import org.joda.time.LocalDate;
+import org.joda.time.Period;
+import org.motechproject.commons.date.model.DayOfWeek;
+import org.motechproject.commons.date.model.Time;
+import org.motechproject.event.MotechEvent;
+import org.motechproject.scheduler.contract.CronSchedulableJob;
+import org.motechproject.scheduler.contract.DayOfWeekSchedulableJob;
+import org.motechproject.scheduler.contract.RepeatingPeriodSchedulableJob;
+import org.motechproject.scheduler.contract.RepeatingSchedulableJob;
+import org.motechproject.scheduler.contract.RunOnceSchedulableJob;
+import org.motechproject.scheduler.contract.JobDto;
+
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.motechproject.scheduler.constants.SchedulerConstants.CRON_EXPRESSION;
+import static org.motechproject.scheduler.constants.SchedulerConstants.DAYS;
+import static org.motechproject.scheduler.constants.SchedulerConstants.END_DATE;
+import static org.motechproject.scheduler.constants.SchedulerConstants.IGNORE_PAST_FIRES_AT_START;
+import static org.motechproject.scheduler.constants.SchedulerConstants.REPEAT_COUNT;
+import static org.motechproject.scheduler.constants.SchedulerConstants.REPEAT_INTERVAL_IN_SECONDS;
+import static org.motechproject.scheduler.constants.SchedulerConstants.REPEAT_PERIOD;
+import static org.motechproject.scheduler.constants.SchedulerConstants.TIME;
+import static org.motechproject.scheduler.constants.SchedulerConstants.USE_ORIGINAL_FIRE_TIME_AFTER_MISFIRE;
+import static org.motechproject.scheduler.util.CustomDateParser.parseToDateTime;
+import static org.motechproject.scheduler.util.CustomDateParser.parseToLocalDate;
+
+public final class SchedulableJobBuilder {
+
+    public static CronSchedulableJob buildCronSchedulableJob(JobDto dto) throws ParseException {
+        Map<String, String> parameters = dto.getParameters();
+
+        MotechEvent event = new MotechEvent(dto.getMotechEventSubject(), dto.getMotechEventParameters());
+        String cronExpression = parameters.get(CRON_EXPRESSION);
+        DateTime startDate = dto.getStartDate();
+        DateTime endDate = parseToDateTime(parameters.get(END_DATE));
```

Comment:
Don't like the idea of storing parameters in a map. This is an object oriented language, can't we use inheritance for this?